### PR TITLE
Jenkins OverwriteConfig setting also overwrites init scripts

### PR DESCRIPTION
**Pull request by wilrnh**
#### What this PR does / why we need it:
This PR replaces existing init scripts in Jenkins when the OverwriteConfig chart setting is enabled.

#### Special notes for your reviewer:
Unsure if to make this an entirely new setting, eg OverwriteInitscripts, if so, happy to PR that instead.

#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [x] [DCO](https://www.helm.sh/blog/helm-dco/index.html) signed
- [x] Chart Version bumped
- [x] Variables are documented in the README.md

### CODE REVIEW

**Comment by wilrnh** [comment]
hey @viglesiasce and @lachie83 any feedback on this pr would be greatly appreciated. thx.

**Comment by maorfr** [comment]
Hey @wilrnh,

Thanks for contributing!
This chart has too many config options as it is, so let's use `OverwriteConfig` only, as you suggest.

If you resolve conflicts we can get this merged ;)

**Comment by maorfr** [comment]
and while you're at it, would you mind taking a look at helm/charts#10372 please?
Thanks!

**Comment by JeanMertz** [comment]
I had a need for this change, and some others related to this, so I created a new PR with those changes.

See: https://github.com/helm/charts/pull/11200

File: stable/jenkins/templates/config.yaml [hunk 1]
```diff
@@ -243,6 +243,9 @@ data:
 {{- end }}
 {{- if .Values.Master.InitScripts }}
     mkdir -p /var/jenkins_home/init.groovy.d/;
+    {{- if .Values.Master.OverwriteConfig }}
```
Comment (maorfr) [inline stable/jenkins/templates/config.yaml]: The indentation here seems wrong

File: stable/jenkins/templates/config.yaml [hunk 2]
```diff
@@ -243,6 +243,9 @@ data:
...
+      rm -f /var/jenkins_home/init.groovy.d/*.groovy
```
Comment (maorfr) [inline stable/jenkins/templates/config.yaml]: The indetation here seems wrong

**Comment by maorfr** [comment]
And as soon as you resolve the conflicts and resolve the comment, we can merge.

This is a great first contribution :)

**Comment by maorfr** [comment]
ping @wilrnh

File: stable/jenkins/templates/config.yaml (hunk 2 above)
Comment (maorfr) [reply stable/jenkins/templates/config.yaml]: this is still over indented (by 2)

Comment (wilrnh) [reply stable/jenkins/templates/config.yaml]: "think" i fixed it here. At least it looks aok with `helm template`, but lemme know 👍

**Comment by wilrnh** [comment]
(fixed dco)

**Comment by maorfr** [comment]
/lgtm

Thanks for contributing!

**Pull request merged**